frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,491 @@
+            onChange={handleChange}
+            aria-describedby="receiver-name-help"
+            name="receiverName"
+            data-test-id="receiver-name"
+            required
+          />
+        </div>
+        <div className="form-group co-m-pane__dropdown">
+          <label className="control-label co-required">Receiver Type</label>
+          <Dropdown
+            title={'Select Receiver Type...'}
+            name="receiverType"
+            items={receiverTypes}
+            dropDownClassName="dropdown--full-width"
+            data-test-id="receiver-type"
+            selectedKey={formValues.receiverType}
+            onChange={(e) => handleChange({ target: { name: 'receiverType', value: e } })}
+          />
+        </div>
+
+        {formValues.receiverType && (
+          <>
+            <SubForm.Form
+              globals={allGlobals}
+              formValues={formValues}
+              handleChange={handleChange}
+            />
+            <RoutingLabelEditor
+              formValues={formValues}
+              setFormValues={setFormValues}
+              isDefaultReceiver={isDefaultReceiver}
+            />
+          </>
+        )}
+
+        <ButtonBar errorMessage={errorMsg} inProgress={inProgress}>
+          <ActionGroup className="pf-c-form">
+            <Button
+              type="submit"
+              variant="primary"
+              data-test-id="save-changes"
+              isDisabled={isFormInvalid as boolean}
+            >
+              {saveButtonText}
+            </Button>
+            <Button
+              type="button"
+              variant="secondary"
+              data-test-id="cancel"
+              onClick={history.goBack}
+            >
+              Cancel
+            </Button>
+          </ActionGroup>
+        </ButtonBar>
+      </form>
+    </div>
+  );
+};
+
+export const SaveAsDefaultCheckbox = ({ formField, disabled, label, formValues, handleChange }) => {
+  const saveAsDefaultLabelClass = classNames({ 'co-no-bold': disabled });
+  return (
+    <label className={saveAsDefaultLabelClass}>
+      <input
+        type="checkbox"
+        name={formField}
+        data-test-id="save-as-default"
+        onChange={(e) =>
+          handleChange({
+            target: { name: `${formField}`, value: e.target.checked },
+          })
+        }
+        checked={formValues[`${formField}`]}
+        disabled={disabled}
+      />
+      &nbsp; {label}
+    </label>
+  );
+};
+
+const ReceiverWrapper: React.FC<ReceiverFormsWrapperProps> = React.memo(({ obj, ...props }) => {
+  const { alertManagerBaseURL } = window.SERVER_FLAGS;
+  const [alertManagerGlobals, setAlertManagerGlobals] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMsg, setErrorMsg] = React.useState('');
+
+  if (_.isEmpty(errorMsg)) {
+    if (!alertManagerBaseURL) {
+      setErrorMsg(`Error alertManagerBaseURL not set`);
+      return;
```

Comment:
This is an early return for the component, so the error message will not be displayed.

Reply:
It does get displayed:
![image](https://user-images.githubusercontent.com/12733153/70637208-db205e80-1c04-11ea-869f-900210590d9c.png)
via [here](https://github.com/openshift/console/blob/446148a9eebea9b075560368bb383788a8ddaf44/frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx#L440)
I believe setting the state reloads component.

Reply:
Thanks for the explanation. In cases like this, I think it's more typically to return the error message directly rather than trigger a state change, but I'd like to hear @spadgett's opinion on this.

Reply:
yes, but some of the errors are asynchronous, so just using the same error mechanize.

Reply:
Changed to `useEffect` and was able to remove some of the logic.